File: OpenWrap/Basic/OtherAdServer/ObjectiveC/SampleApp/SampleApp/Viewcontrollers/DummyAdServer/InBannerVideo/IBVideoViewController.m


#import "IBVideoViewController.h"
#import <POBBannerView.h>
#import "CustomBannerEventHandler.h"

#define DUMMY_AD_SERVER_ADUNIT_ID  @"OtherASBannerAdUnit"
#define OW_ADUNIT_ID  @"OtherASBannerAdUnit"

#define PUB_ID          @"156276"
#define PROFILE_ID      @1757

@interface IBVideoViewController ()<POBBannerViewDelegate>
@property (nonatomic) POBBannerView *bannerView;
@end

@implementation IBVideoViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Create a banner custom event handler for your ad server. Make sure you use
    // separate event handler objects to create each banner view.
    // For example, The code below creates an event handler for DummyAdServer.
    CGSize size300x250 = CGSizeMake(300, 250);
    NSArray *sizes = @[[NSValue valueWithCGSize:size300x250]];
    
    CustomBannerEventHandler *eventHandler = [[CustomBannerEventHandler alloc] initWithAdUnit:DUMMY_AD_SERVER_ADUNIT_ID sizes:sizes];
    
    // Create a banner view
    // For test IDs refer - https://community.pubmatic.com/x/IAI5AQ#TestandDebugYourIntegration-TestProfile/Placement
    self.bannerView = [[POBBannerView alloc] initWithPublisherId:PUB_ID
                                                       profileId:PROFILE_ID
                                                        adUnitId:OW_ADUNIT_ID
                                                         eventHandler:eventHandler];
    // Set the delegate
    self.bannerView.delegate = self;
    
    // Add the banner view to your view hierarchy
    [self addBannerToView:self.bannerView withSize:CGSizeMake(300, 250)];
    
    // Load Ad
    [self.bannerView loadAd];
}

- (void)addBannerToView:(UIView *)bannerView withSize:(CGSize )size{
    bannerView.frame = CGRectMake((CGRectGetWidth(self.view.bounds)-size.width)/2,
                                       CGRectGetHeight(self.view.bounds)-size.height,
                                       size.width, size.height);
    bannerView.autoresizingMask = (UIViewAutoresizingFlexibleTopMargin |
                                   UIViewAutoresizingFlexibleLeftMargin |
                                   UIViewAutoresizingFlexibleRightMargin);
    [self.view addSubview:bannerView];
}

#pragma mark - Banner view delegate methods
//Provides a view controller to use for presenting model views
- (UIViewController *)bannerViewPresentationController {
    return self;
}

// Notifies the delegate that an ad has been successfully loaded and rendered.
- (void)bannerViewDidReceiveAd:(POBBannerView *)bannerView {
    NSLog(@"Banner : Ad received with size %@ ", bannerView.creativeSize);
}

// Notifies the delegate of an error encountered while loading or rendering an ad.
- (void)bannerView:(POBBannerView *)bannerView
didFailToReceiveAdWithError:(NSError *_Nullable)error {
    NSLog(@"Banner : Ad failed with error : %@", [error localizedDescription]);
}

// Notifies the delegate whenever current app goes in the background due to user click
- (void)bannerViewWillLeaveApplication:(POBBannerView *)bannerView {
    NSLog(@"Banner : Will leave app");
}

// Notifies the delegate that the banner ad view will launch a modal on top of the current view controller, as a result of user interaction.
- (void)bannerViewWillPresentModal:(POBBannerView *)bannerView {
    NSLog(@"Banner : Will present modal");
}

// Notifies the delegate that the banner ad view has dismissed the modal on top of the current view controller.
- (void)bannerViewDidDismissModal:(POBBannerView *)bannerView {
    NSLog(@"Banner : Dismissed modal");
}

#pragma mark - dealloc
- (void)dealloc {
    _bannerView = nil;
}

@end

File: OpenWrap/Basic/OtherAdServer/Swift/SampleApp/SampleApp/ViewControllers/DummyAdServer/InBannerVideo/IBVideoViewController.swift


import UIKit

class IBVideoViewController: UIViewController,POBBannerViewDelegate {

    let owAdUnit  = "OtherASBannerAdUnit"
    let dummyAdServerAdUnit  = "OtherASBannerAdUnit"
    let pubId = "156276"
    let profileId: NSNumber = 1757

    var bannerView: POBBannerView?

    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Create a banner custom event handler for your ad server. Make sure you use
        // separate event handler objects to create each banner view.
        // For example, The code below creates an event handler for DummyAdServer.
        let size300x250 = CGSize.init(width: 300, height: 250)
        let sizes =  [NSValue.init(cgSize: size300x250)]
    
        let eventHandler = CustomBannerEventHandler(dummyAdServerAdUnit, sizes: sizes)
        // Create a banner view
        // For test IDs refer - https://community.pubmatic.com/x/IAI5AQ#TestandDebugYourIntegration-TestProfile/Placement
        self.bannerView = POBBannerView(publisherId: pubId, profileId: profileId, adUnitId: owAdUnit, eventHandler: eventHandler)
        // Set the delegate
        self.bannerView?.delegate = self

        // Add the banner view to your view hierarchy
        addBannerToView(banner: self.bannerView!, adSize: CGSize(width: 300, height: 250))
                
        // Load Ad
        self.bannerView?.loadAd()
    }
    
    func addBannerToView(banner : POBBannerView?, adSize : CGSize) -> Void {
        
        banner?.frame = CGRect(x: (self.view.bounds.size.width - adSize.width)/2
            , y: self.view.bounds.size.height - adSize.height, width: adSize.width, height: adSize.height)
        banner?.autoresizingMask = [.flexibleTopMargin, .flexibleLeftMargin, .flexibleRightMargin]
        view.addSubview(banner!)
    }
    
    // MARK: - Banner view delegate methods
    //Provides a view controller to use for presenting modal views
    func bannerViewPresentationController() -> UIViewController {
        return self
    }
    // Notifies the delegate that an ad has been successfully loaded and rendered.
    func bannerViewDidReceiveAd(_ bannerView: POBBannerView) {
        print("Banner : Ad received with size \(String(describing: bannerView.creativeSize())) ")
    }
    
    // Notifies the delegate of an error encountered while loading or rendering an ad.
    func bannerView(_ bannerView: POBBannerView,
                           didFailToReceiveAdWithError error: Error?) {
        print("Banner : Ad failed with error : \(error?.localizedDescription ?? "")")
    }
    
    // Notifies the delegate whenever current app goes in the background due to user click
    func bannerViewWillLeaveApplication(_ bannerView: POBBannerView) {
        print("Banner : Will leave app")
    }
    
    // Notifies the delegate that the banner ad view will launch a modal on top of the current view controller, as a result of user interaction.
    func bannerViewWillPresentModal(_ bannerView: POBBannerView) {
        print("Banner : Will present modal")
    }
    // Notifies the delegate that the banner ad view has dismissed the modal on top of the current view controller.
    func bannerViewDidDismissModal(_ bannerView: POBBannerView) {
        print("Banner : Dismissed modal")
    }

    deinit {
        bannerView = nil
    }
}
